# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.cpp
# Commit: 048c2bf91290
# Full Hash: 048c2bf9129090d91f2111579e130d913c871cdd
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 1
# Description:
#   Bug 1561521 - Clean up wasm::ToString and drop wasm::ToCString. r=lth
#   
#   wasm::ToCString is losing relevance as we have more constructed types. This
#   commit removes ToCString and replaces all uses with wasm::ToString.
#   
# ==============================================================================

diff -r a72132acc190 -r 048c2bf91290 js/src/wasm/WasmJS.cpp
--- a/js/src/wasm/WasmJS.cpp	Tue Aug 18 16:59:22 2020 +0000
+++ b/js/src/wasm/WasmJS.cpp	Tue Aug 18 16:59:26 2020 +0000
@@ -1161,8 +1161,8 @@
       return nullptr;
     }
 
-    const char* argStr = ToCString(arg);
-    if (!buf.append(argStr, strlen(argStr))) {
+    UniqueChars argStr = ToString(arg);
+    if (!buf.append(argStr.get(), strlen(argStr.get()))) {
       return nullptr;
     }
 
@@ -1179,8 +1179,8 @@
       return nullptr;
     }
 
-    const char* resultStr = ToCString(result);
-    if (!buf.append(resultStr, strlen(resultStr))) {
+    UniqueChars resultStr = ToString(result);
+    if (!buf.append(resultStr.get(), strlen(resultStr.get()))) {
       return nullptr;
     }
 